### PR TITLE
feat(new syntax): do not require mode: taint for taint rule

### PR DESCRIPTION
**Pull request by brandonspark**
## What:
This PR makes it so that we don't require `mode: taint` for a taint rule in the new syntax.

## Why:
The top level `taint` key is sufficient to infer that the user wants to input a taint rule.

## How:
Altered the JSON schema and rule parsing a little.

## Test plan:
`make test`

Come to think of it, we probably don't really need it for old-style taint rules either.

No changelog because we're still trying to fly under the radar, kind of.

Closes PA-2608

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: src/parsing/Parse_rule.ml [hunk 1]
```diff
@@ -851,7 +851,7 @@ let find_formula env (rule_dict : dict) : key * G.expr =
   | None ->
       error env rule_dict.first_tok
         "Expected one of `pattern`, `pattern-either`, `patterns`, \
-         `pattern-regex`, `pattern-comby` to be present"
+         `pattern-regex`, `pattern-coby` to be present"
```
Comment (aryx) [inline src/parsing/Parse_rule.ml]: why this change?

**Review by aryx — APPROVED** [review]

File: src/parsing/Parse_rule.ml (hunk 1 above)
Comment (brandonspark) [reply src/parsing/Parse_rule.ml]: that was a mistake from when I was testing why the output wasn't what I expected, good catch

**Comment by brandonspark** [comment]
(sorry for the rerequest, i didn't see you approved)

File: cli/tests/e2e/rules/syntax/good_new_syntax_taint.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+rules:
```
Comment (aryx) [inline cli/tests/e2e/rules/syntax/good_new_syntax_taint.yaml:1]: I still don't understand why this PR shows modification to semgrep_interfaces with lots of
modifications in it that have already been merged.

**Review by aryx — APPROVED** [review]

File: cli/tests/e2e/rules/syntax/good_new_syntax_taint.yaml (hunk 2 above)
Comment (brandonspark) [reply cli/tests/e2e/rules/syntax/good_new_syntax_taint.yaml:1]: cause i didnt merge develop, i guess the base was comparing to a bit older of a version
